pinot-controller/src/main/java/org/apache/pinot/controller/validation/RealtimeSegmentValidationManager.java

```diff
@@ -117,8 +117,18 @@ protected void processTable(String tableNameWithType, Context context) {
 
   private void runSegmentLevelValidation(TableConfig tableConfig, PartitionLevelStreamConfig streamConfig) {
     String realtimeTableName = tableConfig.getTableName();
+
     List<SegmentZKMetadata> segmentsZKMetadata = _pinotHelixResourceManager.getSegmentsZKMetadata(realtimeTableName);
 
+    // Delete tmp segments
```

Comment:
Why put deletion in the segmentLevelValidation? tmp segments deletion is at table level. Sounds like a misfit.

Reply:
It's Jackie's suggestion to avoid potential data race issue if we separate them in different jobs. Check the conversation history.